### PR TITLE
chore: update sphinx

### PR DESCRIPTION
**Pull request by ssbarnea**
- preparing for adding support for py3.10, as we will need sphinx 4.2.0 or newer.
- fix broken docs jobs due to incorrect URL

~Related: https://github.com/pypa/packaging.python.org/issues/995~
Related: https://github.com/ansible-community/devtools/issues/6

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -70,9 +70,11 @@
 
 intersphinx_mapping = {
     'python': ('https://docs.python.org/3', None),
-    'packaging': ('https://packaging.python.org/en/latest/', None),
+    'packaging': ('https://packaging.python.org/', None),
```
Comment (webknjaz) [inline docs/conf.py:73]: This name would be more precise.
```suggestion
    'PyPUG': ('https://packaging.python.org', None),
```

File: docs/conf.py [hunk 2]
```diff
@@ -70,9 +70,11 @@
...
 }
 
+# https://github.com/pypa/packaging.python.org/issues/995
```
Comment (webknjaz) [inline docs/conf.py:76]: This comment seems misplaced. Are you confusing the documentation site dedicated to the Python Packaging Guides vs the library called `packaging`?
```suggestion
```

File: docs/conf.py [hunk 3]
```diff
@@ -70,9 +70,11 @@
...
+nitpick_ignore = [('py:class', 'packaging.version.Version')]
```
Comment (webknjaz) [inline docs/conf.py:77]:
```suggestion
nitpick_ignore = [
    ('py:class', 'packaging.version.Version'),  # autodoc autolinks type annotations, this one is not added
]
```

File: docs/conf.py (hunk 1 above)
Comment (webknjaz) [reply docs/conf.py:73]: Looking closer, I think the correct fix would be to apply
```suggestion
    'packaging': ('https://packaging.pypa.io/en/latest', None),
```
(and maybe drop the ignore)

File: docs/conf.py [hunk 4]
```diff
@@ -70,9 +70,13 @@
...
 intersphinx_mapping = {
     'python': ('https://docs.python.org/3', None),
-    'packaging': ('https://packaging.python.org/en/latest/', None),
+    'packaging': ('https://packaging.python.org/', None),
 }
 
+# https://github.com/pypa/packaging.python.org/issues/995
+nitpick_ignore = [
+    ('py:class', 'packaging.version.Version'),  # autodoc autolinks type annotations, this one is not added
+]
```
Comment (webknjaz) [inline docs/conf.py:79]:
```suggestion
```

File: docs/conf.py [hunk 5]
```diff
@@ -70,9 +70,13 @@
 
 intersphinx_mapping = {
     'python': ('https://docs.python.org/3', None),
-    'packaging': ('https://packaging.python.org/en/latest/', None),
+    'packaging': ('https://packaging.python.org/', None),
 }
 
+# https://github.com/pypa/packaging.python.org/issues/995
+nitpick_ignore = [
+    ('py:class', 'packaging.version.Version'),  # autodoc autolinks type annotations, this one is not added
```
Comment (webknjaz) [inline docs/conf.py:78]: This exists as https://packaging.pypa.io/en/latest/version.html#packaging.version.Version

**Review by webknjaz — APPROVED** [review]

File: docs/conf.py [hunk 6]
```diff
@@ -70,7 +70,8 @@
 
 intersphinx_mapping = {
     'python': ('https://docs.python.org/3', None),
-    'packaging': ('https://packaging.python.org/en/latest/', None),
+    'packaging-guide': ('https://packaging.python.org/', None),
```
Comment (webknjaz) [inline docs/conf.py:73]: is this really used anywhere?

**Review by webknjaz — COMMENTED** [review]
I cannot approve of this title. It explains nothing.

**Review by webknjaz — CHANGES_REQUESTED** [review]
urgh.. why is it so hard to withdraw an approval?

**Comment by ssbarnea** [comment]
> I cannot approve of this title. It explains nothing.

Done, make it explicit.

**Review by webknjaz — APPROVED** [review]

**Pull request merged**